cluster/hooks.go

```diff
@@ -993,3 +1005,84 @@ func addLabelsToNode(client *kubernetes.Clientset, nodeName string, labels map[s
 	_, err = client.CoreV1().Nodes().Patch(nodeName, types.MergePatchType, []byte(patch))
 	return
 }
+
+// TaintHeadNodes add taints to the given node in nodepool
+func TaintHeadNodes(input interface{}) error {
+
+	headNodePoolName := viper.GetString(pipConfig.PipelineHeadNodePoolName)
+	if len(headNodePoolName) == 0 {
+		log.Infof("headNodePoolName not specified")
+		return nil
+	}
+	log.Infof("taint nodes in pool: %v", headNodePoolName)
+
+	commonCluster, ok := input.(CommonCluster)
+	if !ok {
+		return errors.Errorf("Wrong parameter type: %T", commonCluster)
+	}
+
+	log.Debug("get K8S config")
+	kubeConfig, err := commonCluster.GetK8sConfig()
+	if err != nil {
+		return err
+	}
+
+	log.Debug("get K8S connection")
+	client, err := helm.GetK8sConnection(kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	clusterDetails, err := commonCluster.GetClusterDetails()
+	if err != nil {
+		return err
+	}
+	nodePoolDetails := clusterDetails.NodePools[headNodePoolName]
+	if nodePoolDetails == nil {
+		return errors.Errorf("Wrong pool name: %v, configured as head node pool")
+	}
+
+	taintedNodesCount, err := taintNodepoolNodes(client, viper.GetString(pipConfig.PipelineHeadNodePoolName))
+	if err != nil {
+		return err
+	}
+	if taintedNodesCount != nodePoolDetails.Count {
+		log.Warnf("Head node pool configured size (%v) and tainted nodes count (%v) is different", nodePoolDetails.Count, taintedNodesCount)
+	}
+	log.Infof("taint nodes finished")
+
+	return nil
+}
+
+func taintNodepoolNodes(client *kubernetes.Clientset, nodePoolName string) (taintedNodesCount int, err error) {
+	selector := fmt.Sprintf("%s=%s", pkgCommon.LabelKey, nodePoolName)
+	nodes, err := client.CoreV1().Nodes().List(metav1.ListOptions{
+		LabelSelector: selector,
+	})
+	if err != nil {
+		return
```

Comment:
Please set err unless this will returns a `nil`. In `client.CoreV1().Nodes().List(metav1.ListOptions` line the `:=` operator creates a new `err` variable and the returned `err` variable is nil.